lib/trackler/problem.rb

```diff
@@ -7,81 +7,108 @@ class Problem
     def initialize(slug, root)
       @slug = slug
       @root = root
+      @xcommon_repository_url = 'https://github.com/exercism/x-common/blob/master'
     end
 
     def exists?
-      !!md && !!yaml
+      description_exists? && yaml_exists?
     end
 
     def name
-      @name ||= slug.split('-').map(&:capitalize).join(' ')
+      @name ||= slug_to_name
```

Comment:
> I suspect that we have very different views about what is readable code. I prefer to not extract a method if the name of the method is basically an echo of the implementation.

A brief discussion about my opinion on using a more literate programming style:

If I'm looking over the code of the public methods of a class I'd prefer NOT to
see or have to care about implementation details until I really have to.

When I see: `@name ||= slug_to_name` I know that the name is made from the slug
somehow but I don't need to care how, if I do want or need to know I can look it up.
If we're memoizing, it must be slow or somehow important.

When I see: `@name ||= slug.split('-').map(&:capitalize).join(' ')` I see that the name starts with the slug and then has a bunch of stuff done to it, it splits it, captializes some stuff, joins it together again. That's a lot more to think about.  And why are we memoizing? That doesn't look slow or important!?

If it wasn't being memoized there's less reason to do it. 
```ruby
def name
  slug_to_name
end
```
A method that only calls another method is redundant.


And I guess
```
def name
  slug.split('-').map(&:capitalize).join(' ')
end
```
would be fine.


> I prefer to not extract a method if the name of the method is basically an echo of the implementation.

I agree with this, I think our difference is where we draw the line.  

```ruby
def split_on_hyphen(string)
  string.split('-')
end
```
Is certainly too far.

It seems you think it's better to stop earlier.
What is the cost of having more, smaller functions?